Src/FluentAssertions/Equivalency/Steps/EqualityComparerEquivalencyStep.cs

```diff
@@ -21,6 +21,11 @@ public EqualityComparerEquivalencyStep(IEqualityComparer<T> comparer)
             return EquivalencyResult.ContinueWithNext;
         }
 
+        if (comparands.Subject is null && typeof(T).IsValueType)
```

Comment:
🤔 I don't understand why this solves the problem. Care to elaborate?

Reply:
After I wrote this I tried the same issue with a class instead of a struct. It's the same error. So I will add another test for this case and remove the `typeof(T).IsValueType)`.

Reply:
I have added the additional tests and also a extended solution for other null cases. I also added a comment in code to explain the reason of the condition.

Please feel free to provide feedback if any part of my explanation is unclear.